metrotranscom/doorway
# ci: enable codepipeline

## Issue Overview

This PR addresses https://github.com/metrotranscom/doorway-infra/issues/64

Before we merge this, we'll want to change the `ci/buildspec*` files, replacing this branch name, `cicd-test` with `main`.  We'll also want to build the dev infra per running [this terraform](https://github.com/metrotranscom/doorway-infra/tree/main/app-codepipeline) with the `dev` workspace, using the vars [here](https://github.com/metrotranscom/doorway-config/tree/main/app-codepipeline)


## Description

This creates the ci/buildspec* files that the ci/cd pipeline will consume. It also resolves some dependencies needed for building docker images. 

## How Can This Be Tested/Reviewed?

Force pushing to this branch will trigger the codepipeline named `doorway-drumsound-codepipeline` in AWS. The pipeline will succeed and finally deploy to the dev ECS infrastructure. See note about about changing the buildspec file names before merging. 


## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
